lib/app/routes/app_router.dart

```diff
@@ -117,23 +116,51 @@ class AppRouter {
           child: const LoadingPage(),
         ),
       ),
-      GoRoute(
-        path: Pages.generosityIntroduction.path,
-        name: Pages.generosityIntroduction.name,
-        builder: (context, state) => const GenrosityIntorductionScreen(),
-      ),
       GoRoute(
         path: Pages.generosityChallenge.path,
         name: Pages.generosityChallenge.name,
         builder: (context, state) {
           return BlocProvider(
             create: (_) => GenerosityChallengeCubit(
               getIt(),
+              getIt(),
             )..loadFromCache(),
             child: const GenerosityChallenge(),
           );
         },
         routes: [
+          GoRoute(
+            path: Pages.generosityChallengeIntroduction.path,
+            name: Pages.generosityChallengeIntroduction.name,
+            builder: (context, state) {
+              final challengeCubit = state.extra! as GenerosityChallengeCubit;
+              return BlocProvider.value(
+                value: challengeCubit,
+                child: const GenerosityChallengeIntorduction(),
```

Comment:
There is no need for a BlocProvider here

Reply:
I think we do need it, since we use it inside the introduction page here:
`context.goNamed(
                  Pages.generosityChallengeChat.name,
                  extra: context.read<GenerosityChallengeCubit>(),
                );`

Reply:
Ah that's tricky, because the redirection to this page happens within _redirectFromExternalLink and we are not providing GenerosityChallengeCubit through there state.extra is null and i get a red exception screen when i get to this page currently. Funny enough when i removed the GenerosityChallengeCubit provider everything still worked as expected in the chat

Reply:
changed based on talk with Daniela